Docker Image available?

Is there any docker image available?

> @a0v0 What is the use-case/need for that? I mean it is a go library that can be imported via go modules!

1. No headache when installing on different platforms/os. I am getting this error while installing on gitpod.

```log
gitpod@a0v0-mtlssbytr5q /w/ (feat/user-onboarding)> go install github.com/parvez3019/go-swagger3@latest
go: downloading github.com/parvez3019/go-swagger3 v0.0.0-20230312192607-60c9e86c2661
go: downloading github.com/sirupsen/logrus v1.8.1
go: downloading github.com/mikunalpha/go-module v0.0.0-20190521120234-12aa2dc244ca
go: downloading github.com/iancoleman/orderedmap v0.2.0
go: downloading github.com/ghodss/yaml v1.0.0
go: downloading github.com/stretchr/testify v1.7.0
go: downloading github.com/urfave/cli v1.22.5
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading golang.org/x/sys v0.0.0-20191026070338-33540a1f6037
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/stretchr/objx v0.1.0
go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.0-20190314233015-f79a8a8ca69d
go: downloading gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c
go: downloading github.com/russross/blackfriday/v2 v2.0.1
go: downloading github.com/shurcooL/sanitized_anchor_name v1.0.0
# crypto/internal/nistec
/home/gitpod/go/src/crypto/internal/nistec/p256_ordinv.go:14:6: p256OrdMul redeclared in this block
        /home/gitpod/go/src/crypto/internal/nistec/p256_asm_ordinv.go:14:6: other declaration of p256OrdMul
/home/gitpod/go/src/crypto/internal/nistec/p256_ordinv.go:19:6: p256OrdSqr redeclared in this block
        /home/gitpod/go/src/crypto/internal/nistec/p256_asm_ordinv.go:19:6: other declaration of p256OrdSqr
/home/gitpod/go/src/crypto/internal/nistec/p256_ordinv.go:21:6: P256OrdInverse redeclared in this block
        /home/gitpod/go/src/crypto/internal/nistec/p256_asm_ordinv.go:21:6: other declaration of P256OrdInverse
```
2. A docker version would cast away all the installation problems like this and keep the local development env clean. Also updating with latest version is lot easier.



here is the dockerfile I wrote, if it helps
```
# Use the latest Golang image as the base image
FROM golang:latest AS builder

# Install Git to download Go packages
RUN apt-get update && \
    apt-get install -y git && \
    rm -rf /var/lib/apt/lists/*

# Download and install go-swagger3
RUN go install github.com/parvez3019/go-swagger3@latest

# Create a new image with only the binary and runtime dependencies
FROM debian:latest
COPY --from=builder /go/bin/go-swagger3 /usr/local/bin/go-swagger3
ENTRYPOINT ["go-swagger3"]
```

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -16,9 +16,6 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
         
-      - name: Extract Git Tag
-        run: echo "GIT_TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
-
       - name: Login to Docker Hub
         uses: docker/login-action@v1
         with:
@@ -33,12 +30,4 @@ jobs:
           push: true
           tags: |
             ${{ env.DOCKER_USERNAME }}/go-swagger3:latest
-            # ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}
             ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ env.GIT_TAG }}
-            
-      - name: Get Docker Image Digest
-        id: docker_digest
-        run: echo "::set-output name=digest::$(docker inspect --format='{{index .RepoDigests 0}}' '${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}')"
-
-      - name: Display Image Digest
-        run: echo ${{ steps.docker_digest.outputs.digest }}

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -3,7 +3,7 @@ name: Docker Build and Publish
 on:
   push:
     tags:
-      - v*
+      - "*"
 
 env:
   DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
@@ -15,6 +15,9 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
+        
+      - name: Extract Git Tag
+        run: echo "GIT_TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
 
       - name: Login to Docker Hub
         uses: docker/login-action@v1
@@ -30,8 +33,9 @@ jobs:
           push: true
           tags: |
             ${{ env.DOCKER_USERNAME }}/go-swagger3:latest
-            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}
-
+            # ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}
+            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ env.GIT_TAG }}
+            
       - name: Get Docker Image Digest
         id: docker_digest
         run: echo "::set-output name=digest::$(docker inspect --format='{{index .RepoDigests 0}}' '${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}')"

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -32,11 +32,6 @@ jobs:
         with:
           username: ${{ env.DOCKER_USERNAME }}
           password: ${{ env.DOCKER_PASSWORD }}
-      
-      - name: somethin
-        run: |
-          echo "Get tag name  ${{ steps.get_tag_name.outputs.tag }}"
-          echo  ${{ steps.meta.outputs.tags }}
           
       - name: Build and tag Docker image
         id: docker_build

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -16,34 +16,34 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
         
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.DOCKER_USERNAME }}/go-swagger3
+          flavor: latest=true
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            
       - name: Login to Docker Hub
         uses: docker/login-action@v1
         with:
           username: ${{ env.DOCKER_USERNAME }}
           password: ${{ env.DOCKER_PASSWORD }}
-
-      - name: Get tag name
-        uses: actions/github-script@v4
-        id: get_tag_name
-        with:
-          script: |
-            const tag = context.payload.ref.replace('refs/tags/', '');
-            return { tag: tag };
-      - name: somethin
-        run: echo "GIT_TAG=`echo $(git describe --tags --abbrev=0)`" >> $GITHUB_ENV
-
+      
       - name: somethin
         run: |
           echo "Get tag name  ${{ steps.get_tag_name.outputs.tag }}"
           echo ${{ env.GIT_TAG }}
-          echo ${{ GIT_TAG }}
+          echo ${{ GIT_TAG }} ${{ steps.meta.outputs.tags }}
           
       - name: Build and tag Docker image
         id: docker_build
         uses: docker/build-push-action@v2
         with:
           context: .
-          push: true
-          tags: |
-            ${{ env.DOCKER_USERNAME }}/go-swagger3:latest
-            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ env.GIT_TAG }}
+          push: ${{ github.ref_type == 'tag' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -29,13 +29,14 @@ jobs:
           script: |
             const tag = context.payload.ref.replace('refs/tags/', '');
             return { tag: tag };
-        
+      - name: somethin
+        run: echo "GIT_TAG=`echo $(git describe --tags --abbrev=0)`" >> $GITHUB_ENV
+
       - name: somethin
         run: |
           echo "Get tag name  ${{ steps.get_tag_name.outputs.tag }}"
-          echo "GIT_TAG=`echo $(git describe --tags --abbrev=0)`" >> $GITHUB_ENV
-          echo "env.GIT_TAG  ${{ env.GIT_TAG }}"
-          echo "GIT_TAG  ${{ GIT_TAG }}"
+          echo ${{ env.GIT_TAG }}
+          echo ${{ GIT_TAG }}
           
       - name: Build and tag Docker image
         id: docker_build

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -36,8 +36,7 @@ jobs:
       - name: somethin
         run: |
           echo "Get tag name  ${{ steps.get_tag_name.outputs.tag }}"
-          echo ${{ env.GIT_TAG }}
-          echo ${{ GIT_TAG }} ${{ steps.meta.outputs.tags }}
+          echo  ${{ steps.meta.outputs.tags }}
           
       - name: Build and tag Docker image
         id: docker_build

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -29,7 +29,14 @@ jobs:
           script: |
             const tag = context.payload.ref.replace('refs/tags/', '');
             return { tag: tag };
-
+        
+      - name: somethin
+        run: |
+          echo "Get tag name  ${{ steps.get_tag_name.outputs.tag }}"
+          echo "GIT_TAG=`echo $(git describe --tags --abbrev=0)`" >> $GITHUB_ENV
+          echo "env.GIT_TAG  ${{ env.GIT_TAG }}"
+          echo "GIT_TAG  ${{ GIT_TAG }}"
+          
       - name: Build and tag Docker image
         id: docker_build
         uses: docker/build-push-action@v2
@@ -38,4 +45,4 @@ jobs:
           push: true
           tags: |
             ${{ env.DOCKER_USERNAME }}/go-swagger3:latest
-            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ steps.get_tag_name.outputs.tag }}
+            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ env.GIT_TAG }}

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -2,8 +2,8 @@ name: Docker Build and Publish
 
 on:
   push:
-    branches:
-      - main
+    tags:
+      - v*
 
 env:
   DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -1,0 +1,40 @@
+name: Docker Build and Publish
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Build and tag Docker image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          tags: |
+            ${{ env.DOCKER_USERNAME }}/go-swagger3:latest
+            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}
+
+      - name: Get Docker Image Digest
+        id: docker_digest
+        run: echo "::set-output name=digest::$(docker inspect --format='{{index .RepoDigests 0}}' '${{ env.DOCKER_USERNAME }}/go-swagger3:${{ github.sha }}')"
+
+      - name: Display Image Digest
+        run: echo ${{ steps.docker_digest.outputs.digest }}

--- a/.github/workflows/docker.yml
+++ b/.github/workflows/docker.yml
@@ -22,6 +22,14 @@ jobs:
           username: ${{ env.DOCKER_USERNAME }}
           password: ${{ env.DOCKER_PASSWORD }}
 
+      - name: Get tag name
+        uses: actions/github-script@v4
+        id: get_tag_name
+        with:
+          script: |
+            const tag = context.payload.ref.replace('refs/tags/', '');
+            return { tag: tag };
+
       - name: Build and tag Docker image
         id: docker_build
         uses: docker/build-push-action@v2
@@ -30,4 +38,4 @@ jobs:
           push: true
           tags: |
             ${{ env.DOCKER_USERNAME }}/go-swagger3:latest
-            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ env.GIT_TAG }}
+            ${{ env.DOCKER_USERNAME }}/go-swagger3:${{ steps.get_tag_name.outputs.tag }}

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,11 @@
+FROM golang:alpine as builder
+WORKDIR /app
+COPY . .
+RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/binary .
+RUN upx -9 /bin/binary
+
+FROM gcr.io/distroless/static:nonroot
+WORKDIR /app/
+COPY --from=builder /bin/binary /bin/binary
+ENTRYPOINT ["/bin/binary"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,28 +1,5 @@
-# FROM golang:alpine as builder
-# WORKDIR /app
-# COPY . .
-# RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
-# RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/binary .
-# RUN upx -9 /bin/binary
+FROM golang:alpine 
+WORKDIR /go/src/main
+RUN go install github.com/parvez3019/go-swagger3@latest
 
-# FROM gcr.io/distroless/static:nonroot
-# WORKDIR /app/
-# COPY --from=builder /bin/binary /bin/binary
-# ENTRYPOINT ["/bin/binary"]
-
-
-
-FROM golang:alpine as builder
-
-WORKDIR /work
-COPY . .
-RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/swagger .
-RUN upx -9 /bin/swagger
-
-FROM gcr.io/distroless/static:nonroot
-
-COPY --from=builder bin/swagger /usr/bin/swagger
-
-ENTRYPOINT ["/usr/bin/swagger"]
-
+ENTRYPOINT ["go-swagger3"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -14,18 +14,15 @@
 
 FROM golang:alpine as builder
 
-ADD . /work
 WORKDIR /work
-
-RUN apk --no-cache add ca-certificates shared-mime-info mailcap git build-base binutils-gold
-
-RUN mkdir -p bin &&\
-  RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o bin/swagger .
+COPY . .
+RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/swagger .
 RUN upx -9 /bin/swagger
 
 FROM golang:alpine
 
-COPY --from=cross /work/bin/swagger /usr/bin/swagger
+COPY --from=builder bin/swagger /usr/bin/swagger
 
 ENTRYPOINT ["/usr/bin/swagger"]
 CMD ["--help"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,11 +1,31 @@
-FROM golang:alpine as builder
-WORKDIR /app
-COPY . .
-RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/binary .
-RUN upx -9 /bin/binary
+# FROM golang:alpine as builder
+# WORKDIR /app
+# COPY . .
+# RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
+# RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/binary .
+# RUN upx -9 /bin/binary
 
-FROM gcr.io/distroless/static:nonroot
-WORKDIR /app/
-COPY --from=builder /bin/binary /bin/binary
-ENTRYPOINT ["/bin/binary"]
+# FROM gcr.io/distroless/static:nonroot
+# WORKDIR /app/
+# COPY --from=builder /bin/binary /bin/binary
+# ENTRYPOINT ["/bin/binary"]
+
+
+
+FROM golang:alpine as builder
+
+ADD . /work
+WORKDIR /work
+
+RUN apk --no-cache add ca-certificates shared-mime-info mailcap git build-base binutils-gold
+
+RUN mkdir -p bin &&\
+  RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o bin/swagger .
+RUN upx -9 /bin/swagger
+
+FROM golang:alpine
+
+COPY --from=cross /work/bin/swagger /usr/bin/swagger
+
+ENTRYPOINT ["/usr/bin/swagger"]
+CMD ["--help"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -20,9 +20,9 @@ RUN apk update && apk add upx ca-certificates openssl && update-ca-certificates
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o /bin/swagger .
 RUN upx -9 /bin/swagger
 
-FROM golang:alpine
+FROM gcr.io/distroless/static:nonroot
 
 COPY --from=builder bin/swagger /usr/bin/swagger
 
 ENTRYPOINT ["/usr/bin/swagger"]
-CMD ["--help"]
+

--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@ go install github.com/parvez3019/go-swagger3@latest
 
 
 ### 2. Documentation Generation
-
+#### Using binary
 Go to the folder where is main.go in
 
 ``` shell
@@ -52,6 +52,21 @@ Notes -
 - Pass generate-yaml as trus if you want to generate yaml spec file instead of json
 
 ```
+
+#### Using docker
+``` shell
+// go.mod and main file are in the same directory
+docker run -t --rm -v $(pwd):/app -w /app parvez3019/go-swagger3:latest --module-path . --output oas.json --schema-without-pkg --generate-yaml true
+
+// go.mod and main file are in the different directory
+docker run -t --rm -v $(pwd):/app -w /app parvez3019/go-swagger3:latest --module-path . --main-file-path ./cmd/xxx/main.go --output oas.json --schema-without-pkg --generate-yaml true
+
+Notes - 
+- Pass schema-without-pkg flag as true if you want to generate schemas without package names
+- Pass generate-yaml as trus if you want to generate yaml spec file instead of json
+
+```
+
 
 
 

